pkg/v1/tarball/layer.go

```diff
@@ -19,25 +19,37 @@ import (
 	"compress/gzip"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"os"
 	"sync"
 
 	"github.com/containerd/stargz-snapshotter/estargz"
 	"github.com/google/go-containerregistry/internal/and"
+	comp "github.com/google/go-containerregistry/internal/compression"
 	gestargz "github.com/google/go-containerregistry/internal/estargz"
 	ggzip "github.com/google/go-containerregistry/internal/gzip"
+	"github.com/google/go-containerregistry/internal/zstd"
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/types"
 )
 
+// LayerCompression is an enumeration of the supported compression algorithms for tarball layers
+type LayerCompression comp.Compression
+
+// The collection of known MediaType values.
+const (
+	None LayerCompression = "none"
```

Comment:
I don't love these symbols being `tarball.None`, etc...

I am tempted to just expose a `compression` package so that we don't have to duplicate these everywhere... @imjasonh wdyt?

Reply:
Yes I agree that `tarball.None` looks weird. I would also be in favor of creating a non-internal `compression` package.

Reply:
I created a public `pkg/compression` package that contains the `Compression` enum.